src/withReadyRouter.tsx

```diff
@@ -0,0 +1,25 @@
+import { useRouter } from 'next/router';
+import React, { useEffect, useState } from 'react';
+
+/**
+ * Ensures that Next's router is always ready (i.e. has
+ * query params loaded)
+ */
+export const withReadyRouter = (WrappedComponent: any) => {
+  WrappedComponent.displayName = `WithReadyRouter${WrappedComponent.displayName}`;
```

Comment:
```suggestion
  WrappedComponent.displayName = `WithReadyRouter(${WrappedComponent.displayName || WrappedComponent.name})`;
```

However, this logic is still somewhat simplistic and won't cover all cases - it's fine for our purposes though.